pkg/config/metadata/user_info.go

```diff
@@ -17,26 +19,58 @@ type UserInfo struct {
 	ClientSecret string `yaml:"client_secret"`
 }
 
-func (self *UserInfo) Call(authContext common.AuthContext, ctx context.Context) (interface{}, error) {
-	// find oidc config and the userinfo endpoint
-	idConfig, _ := authContext.FindIdentityByName(self.OIDC)
+func (userinfo *UserInfo) Call(authContext common.AuthContext, ctx context.Context) (interface{}, error) {
+	// find associated oidc identity config
+	var oidcIdentityConfig *identity.OIDC
+	identityConfig, err := authContext.FindIdentityConfigByName(userinfo.OIDC)
```

Comment:
Another way would be assigning the pointer to the Identity when building the `UserInfo` struct

Reply:
And I think that other way will actually be a better one, but I'll leave it for another PR.

In this next iteration, I want to point the UserInfo struct directly to the OIDC one or allow a user_info endpoint directly in the UserInfo struct alternatively. The tricky part is that we'll soon introduce OAuth identity type which technically would admit a pointing UserInfo struct too.

Either way, if/when the UserInfo struct points to an OIDC identity struct, the identity struct should be caching the discovered OIDC configuration, making possible to save that step done again by `UserInfo.Call`.

Reply:
And I think that other way will actually be a better one, but I'll leave it for another PR.

In this next iteration, I want to point the UserInfo struct directly to the OIDC one or, alternatively, allow a user_info endpoint to be set directly in the UserInfo struct. The tricky part of this is that soon we'll be introducing OAuth identity type which should accept UserInfo structs pointing to as well, so this pointer would have to be to either an OIDC struct or to an OAuth struct. While as a string, we don't have such problem.

Either way, if/when the UserInfo struct points to an identity struct (OIDC or OAuth), this identity struct should hold the endpoints. If the identity config is OIDC, then it will be caching the entire OIDC config actually and this will make possible to save a step in `UserInfo.Call`.